catboost/libs/algo/split.h

```diff
@@ -109,7 +110,7 @@ struct TSplit : public TSplitCandidate {
     inline void Load(IInputStream* s) {
         ::LoadMany(s, static_cast<TSplitCandidate&>(*this), BinBorder);
     }
-    TModelSplit GetModelSplit(const TLearnContext& ctx, const TDataset& learnData) const;
+    TModelSplit GetModelSplit(const TLearnProgress& learnProgress, const TCtrHelper& ctrHelper, const TDataset& learnData) const;
```

Comment:
Why not just get `ctx.LearnProgress` & `ctx.CtrsHelper` inside the function? Why changing the signature?

Reply:
Context object (ctx) is used during training (it's filled with some stuff based on training dataset at least), but we don't have one in our mode, we still want to use this method though. So that's why I reduced the dependency space of the method.